apps/entity-service/src/entities/processors/site.processor.ts

```diff
@@ -1,24 +1,161 @@
+  async findOne(uuid: string) {
+    return await Site.findOne({
+      where: { uuid },
+      include: [
+        { association: "framework" },
+        { association: "project", attributes: ["name"], include: [{ association: "organisation" }] }
+      ]
+    });
   }
 
-  async findMany(): Promise<PaginatedResult<Site>> {
-    throw new Error("Method not implemented.");
+  async findMany(query: EntityQueryDto, userId?: number, permissions?: string[]): Promise<PaginatedResult<Site>> {
+    const builder = await this.entitiesService.buildQuery(Site, query, [
+      { association: "project", attributes: ["name"] },
+      { association: "framework" }
+    ]);
+
+    if (query.sort != null) {
+      if (["name"].includes(query.sort.field)) {
+        builder.order([query.sort.field, query.sort.direction ?? "ASC"]);
+      } else if (query.sort.field === "projectName") {
+        builder.order(["project", "name", query.sort.direction ?? "ASC"]);
+      } else if (query.sort.field !== "id") {
+        throw new BadRequestException(`Invalid sort field: ${query.sort.field}`);
+      }
+    }
+
+    const frameworkPermissions = permissions
+      ?.filter(name => name.startsWith("framework-"))
+      .map(name => name.substring("framework-".length) as FrameworkKey);
+    if (frameworkPermissions?.length > 0) {
+      builder.where({ frameworkKey: { [Op.in]: frameworkPermissions } });
+    } else if (permissions?.includes("manage-own")) {
+      builder.where({
+        projectId: { [Op.in]: ProjectUser.userProjectsSubquery(userId) }
+      });
+    } else if (permissions?.includes("projects-manage")) {
+      builder.where({
+        projectId: { [Op.in]: ProjectUser.projectsManageSubquery(userId) }
+      });
+    }
+
+    for (const term of ["name", "projectName"]) {
+      if (query[term] != null) builder.where({ [term]: query[term] });
+    }
+    if (query.search != null) {
+      builder.where({ name: { [Op.like]: `%${query.search}%` } });
+    }
+
+    return { models: await builder.execute(), paginationTotal: await builder.paginationTotal() };
+  }
+
+  async addFullDto(document: DocumentBuilder, site: Site): Promise<void> {
+    const siteId = site.id;
+
+    const approvedSiteReportsQuery = SiteReport.approvedIdsSubquery({ val: siteId });
+    const seedsPlantedCount = (await Seeding.visible().siteReports(approvedSiteReportsQuery).sum("amount")) ?? 0;
+    const treesPlantedCount =
+      (await TreeSpecies.visible().collection("tree-planted").siteReports(approvedSiteReportsQuery).sum("amount")) ?? 0;
+
+    const hasBeenSubmittedSiteReports = await SiteReport.hasBeenSubmitted()
+      .sites([siteId])
+      .findAll({ attributes: ["id", "siteId", "numTreesRegenerating"] });
+
+    const approvedSiteReports = await SiteReport.approved()
+      .sites([siteId])
+      .findAll({ attributes: ["id", "siteId", "numTreesRegenerating"] });
+
+    const regeneratedTreesCount = sumBy(hasBeenSubmittedSiteReports, "numTreesRegenerating");
```

Comment:
Another spot where the audit I did in the spreadsheet will help you. This field is not used in the FE. I would rename the `approvedRegeneratedTreesCount` to just `regeneratedTreesCount`, and remove this submitted reports field entirely. 

A general guidance from product that I forgot to put on the tickets is that all aggregated fields on these entities should be based only on approved entities. The fact that some of them in PHP are using the submitted entities instead is a mistake.